### ISSUE TITLE
Installation on FreeBSD instruction

### ISSUE THREAD

**Issue by Chinaski1** [AdguardTeam/AdGuardHome#1352]
## How to install:
- pkg install wget ca_root_nss #necessary package installation
- wget https://static.adguard.com/adguardhome/release/AdGuardHome_freebsd_amd64.tar.gz #binary archive download
- tar -xvf AdGuardHome_freebsd_amd64.tar.gz #unzip it in the same folder as the downloaded one.
- chmod -R 755 ./AdGuardHome #confirm rights to execution
- nohup ./AdGuardHome/AdGuardHome & # start the package with the session sh binding
- Go  to the server address
192.168.x.x:3000 - for further configuration
- You can install both on FreeBSD and Jail hypervisor, NAS systems on
Based on FreeBSD: Xigmanas, FreeaNAS.

- In addition, you can automate startup and provide control of
of the activity script.

**Comment by ezoSresyeK** [comment]
>     * pkg install wget ca_root_nss #necessary package installation
> 
>     * wget https://static.adguard.com/adguardhome/release/AdGuardHome_freebsd_amd64.tar.gz #binary archive download

Can simplify further:
* pkg install ca_root_nss
* fetch https://static.adguard.com/adguardhome/release/AdGuardHome_freebsd_amd64.tar.gz

FreeBSD has an in base ftp/http downloader (used by pkg) and it makes little sense to install wget so as to do something that can already be done. Not saying wget isn't a valid option but that its a superfluous package is all.

**Comment by eoghan2t9** [comment]
In case anyone new to FreeBSD has this installed on there system and whats it to start at boot up here is a very simple boot script:

copy this text in a file located in /etc/rc.d/ i.e:

`nano /etc/rc.d/adguardhome
`

```
#!/bin/sh

. /etc/rc.subr

name="adguardhome"
rcvar="adguardhome_enable"
adguardhome_user="root"
adguardhome_command="/root/AdGuardHome/AdGuardHome"
pidfile="/var/run/${name}.pid"
command="/usr/sbin/daemon"
command_args="-P ${pidfile} -r -f ${adguardhome_command}"

load_rc_config $name
: ${adguardhome_enable:=no}

run_rc_command "$1"

```
Change any vars to matchup with your own setup i.e change: 

> adguardhome_command="/root/AdGuardHome/AdGuardHome" 

To match the location you extracted it too. Then make the script executable by:

`chmod +x /etc/rc.d/adguardhome 
`

Don't forget to enable it to start at boot by issuing

`echo "adguardhome_enable='YES'" >> /etc/rc.conf
`

Hope this helps someone out ;)

**Comment by HorayNarea** [comment]
AdGuardHome is also included in the FreeBSD Ports collection so you can simply
`pkg install adguardhome`

**Comment by ameshkov** [comment]
> AdGuardHome is also included in the FreeBSD Ports collection so you can simply

Oh, that's cool! I wonder who maintains the package

**Comment by thoggs** [comment]
> AdGuardHome is also included in the FreeBSD Ports collection so you can simply
> `pkg install adguardhome`

You can't make this door package working properly, you may be missing something, but it doesn't work...

**Comment by ameshkov** [comment]
I wonder can we do something to make packaging possible?

There's an ongoing discussion about Arch Linux packaging: #1324 
Is it relevant?

**Comment by HorayNarea** [comment]
> Oh, that's cool! I wonder who maintains the package

You can see that at https://www.freshports.org/www/adguardhome/ …apparently it's yuri@freebsd.org

**Comment by HorayNarea** [comment]
> > AdGuardHome is also included in the FreeBSD Ports collection so you can simply
> > `pkg install adguardhome`
> 
> You can't make this door package working properly, you may be missing something, but it doesn't work...

well as far as i can tell, the package does not include an rc script so you still need to use that from @eoghan2t9

**Comment by ameshkov** [comment]
Relevant guide: https://github.com/AdguardTeam/AdGuardHome/issues/1703

**Comment by thoggs** [comment]
I understand, but in that case, what kind of maintainer, no?
I think the most interesting thing is to focus on a centralized installation form, like AdGuardHome -s install, something that works on both Linux, Windows, macOS, and FreeBSD ... If you want, you can include that in the source code.

De: Andrey Meshkov <notifications@github.com>
Responder para: AdguardTeam/AdGuardHome <reply@reply.github.com>
Data: terça-feira, 19 de maio de 2020 05:34
Para: AdguardTeam/AdGuardHome <AdGuardHome@noreply.github.com>
Cc: Thiago Rodrigues <thiagothogs@gmail.com>, Comment <comment@noreply.github.com>
Assunto: Re: [AdguardTeam/AdGuardHome] Installation on FreeBSD instruction (#1352)


I wonder can we do something to make packaging possible?

There's an ongoing discussion about Arch Linux packaging: #1324<https://github.com/AdguardTeam/AdGuardHome/issues/1324>
Is it relevant?

**Comment by thoggs** [comment]
Got it, yes, you're right ...
However, I was unable to install AdGuardHome using ports/pkg, in a concise and interactive way, there is no official valid documentation that instructs users to do this, neither from the AdGuardHome team nor from the FreeBSD community.


De: Thomas Sänger <notifications@github.com>
Responder para: AdguardTeam/AdGuardHome <reply@reply.github.com>
Data: terça-feira, 19 de maio de 2020 06:16
Para: AdguardTeam/AdGuardHome <AdGuardHome@noreply.github.com>
Cc: Thiago Rodrigues <thiagothogs@gmail.com>, Comment <comment@noreply.github.com>
Assunto: Re: [AdguardTeam/AdGuardHome] Installation on FreeBSD instruction (#1352)


AdGuardHome is also included in the FreeBSD Ports collection so you can simply
pkg install adguardhome

You can't make this door package working properly, you may be missing something, but it doesn't work...

well as far as i can tell, the package does not include an rc script so you still need to use that from @eoghan2t9<https://github.com/eoghan2t9>

**Comment by ameshkov** [comment]
> I think the most interesting thing is to focus on a centralized installation form, like AdGuardHome -s install, something that works on both Linux, Windows, macOS, and FreeBSD

I actually think it makes sense to support both approaches.

There are a lot of people who prefer using package managers. If we cannot maintain packages on our own, we should at least make it easier for others to do.

What for `-s install` support, we're using https://github.com/kardianos/service/ and it currently does not support `rc.d`

**Comment by thoggs** [comment]
Understand...


De: Andrey Meshkov <notifications@github.com>
Responder para: AdguardTeam/AdGuardHome <reply@reply.github.com>
Data: terça-feira, 19 de maio de 2020 19:01
Para: AdguardTeam/AdGuardHome <AdGuardHome@noreply.github.com>
Cc: Thiago Rodrigues <thiagothogs@gmail.com>, Comment <comment@noreply.github.com>
Assunto: Re: [AdguardTeam/AdGuardHome] Installation on FreeBSD instruction (#1352)


I think the most interesting thing is to focus on a centralized installation form, like AdGuardHome -s install, something that works on both Linux, Windows, macOS, and FreeBSD

I actually think it makes sense to support both approaches.

There are a lot of people who prefer using package managers. If we cannot maintain packages on our own, we should at least make it easier for others to do.

What for -s install support, we're using https://github.com/kardianos/service/ and it currently does not support rc.d

**Comment by ameshkov** [comment]
FreeBSD is supported since v0.103, `-s install` now works as expected

**Comment by conradseba** [comment]
Hi, I'm following the steps, but after changing permissions I still get "./AdGuardHome: Exec format error. Binary file not executable."
I;m on FreeBSD 12.2, pfsense 21.02.2, arm64
Thank you.

**Comment by ainar-g** [comment]
@conradseba, hello.  You probably need the ARM64 binary.  It should be available in v0.107.0.  See #2441.

**Comment by conradseba** [comment]
> #2441

Yeah, sorry, I thought that armv8 was capable of running armv7, my bad. Guess I will have to wait until v0.107.0. Some time in June maybe? thanks!
